feat(AC-9): add invoices page

ralucamavriche opened the pull request:
## Description
feat(AC-9): add invoices page

## How Has This Been Tested?
- [x] Unit tests
- [x] Manual testing

## Screenshots (if applicable)

#### Desktop Page:
<img width="1525" height="741" alt="Screenshot 2025-09-23 at 14 32 00" src="https://github.com/user-attachments/assets/7692ac1d-c587-44a5-8afe-6437374875a5" />

[Screen Recording 2025-09-23 at 14.32.15
](https://github.com/user-attachments/assets/5e983f37-21f2-4107-8320-8dc3a3fd0067)


#### Mobile Page:
<img width="346" height="704" alt="Screenshot 2025-09-23 at 14 49 48" src="https://github.com/user-attachments/assets/e73dd64e-35ff-4d8a-862b-c55a59c90f70" />



<img width="345" height="53" alt="Screenshot 2025-09-23 at 14 49 56" src="https://github.com/user-attachments/assets/3ebdefa7-7112-45de-bb7d-961755938293" />



## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code where necessary
- [x] I have updated the documentation if needed
- [x] My changes generate no new warnings
- [x] New and existing tests pass locally

ralucamavriche committed:
feat: add CreateInvoiceButton component with stories and tests

feat: implement StatusButton component with stories and tests

feat: create CustomerDesktopCard and CustomerMobileCard components for displaying customer data

feat: add Customers component to manage both desktop and mobile views

feat: introduce Pagination component for navigating through invoice lists

feat: add SearchBar component for filtering invoices

feat: enhance InvoicesPage with new components and mock customer data

fix: update formatCurrency utility to handle string inputs

fix: improve formatDate utility to return formatted date strings

GligaDumitru commented on src/components/Pagination/Pagination.tsx:
```diff
@@ -0,0 +1,47 @@
+  return (
+    <div className="mt-5 flex justify-center">
+      <div className="inline-flex">
+        <a
```
use buttons, not `a` elements

GligaDumitru commented on src/components/Pagination/Pagination.tsx:
```diff
@@ -0,0 +1,47 @@
+import { ArrowLeftIcon, ArrowRightIcon } from '@heroicons/react/20/solid';
+
+const Pagination = () => {
```
make this function ready to be used, should accept total, currentPage, onClick callbacks,

GligaDumitru commented on src/components/Pagination/Pagination.tsx:
```diff
@@ -0,0 +1,47 @@
+            aria-current="page"
+            className="z-10 flex h-10 w-10 items-center justify-center rounded-l-md border border-blue-600 bg-blue-600 text-sm text-gray-300"
+          >
+            1
```
Generate the pages dinamic, based on the total prop, and the number items perPage

GligaDumitru commented on src/components/SearchBar/SearchBar.tsx:
```diff
@@ -0,0 +1,24 @@
+import { MagnifyingGlassIcon } from '@heroicons/react/24/outline';
+
+const SearchBar = () => {
```
this should be ready to integrate. Should have props as value and onChange callback. Also take into account about using a debouce function

ralucamavriche committed:
feat: refactor customer components to use unified CustomerDataProps and implement pagination with search functionality

GligaDumitru commented on src/components/Pagination/Pagination.tsx:
```diff
@@ -0,0 +1,68 @@
+    <div className="mt-5 flex justify-center">
+      <div className="inline-flex">
+        <button
+          disabled={currentPage === 1}
```
you are repeating the logic for `currentPage === 1`, please use a variable under useMemo

ralucamavriche replied:
I can improve the repetition by using useMemo if the value is derived from props or state and I want to optimize recalculation. For this simple boolean, I think a local variable is enough.

`const isFirstPage = currentPage === 1;`

GligaDumitru commented on src/components/Pagination/Pagination.tsx:
```diff
@@ -0,0 +1,68 @@
+            <button
+              key={page}
+              onClick={() => onPageChange(page)}
+              aria-current={page === currentPage ? 'page' : undefined}
```
the same with `page === currentPage`

ralucamavriche committed:
feat: enhance Pagination component layout for better responsiveness

GligaDumitru commented on src/components/Pagination/Pagination.tsx:
```diff
@@ -0,0 +1,68 @@
+        </div>
+        <button
+          onClick={onNextPageHandle}
+          disabled={currentPage === totalPages}
```
the same here

GligaDumitru commented on src/components/SearchBar/SearchBar.tsx:
```diff
@@ -0,0 +1,41 @@
+  handleSearch: (query: string) => void;
+}
+
+function debounce(func, timeout = 300) {
```
this should be moved in a util/helper folder

GligaDumitru commented on src/pages/InvoicesPage/InvoicesPage.tsx:
```diff
@@ -12,14 +22,62 @@ const InvoiceMetadata = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const filteredCustomers: Array<Customer> = customers.filter((customer) => {
```
useMemo

GligaDumitru commented on src/pages/InvoicesPage/InvoicesPage.tsx:
```diff
@@ -12,14 +22,62 @@ const InvoiceMetadata = () => {
+      (value) => typeof value === 'string' && value.toLowerCase().includes(query),
+    );
+  });
+  const totalPages: number = Math.ceil(filteredCustomers.length / ITEMS_PER_PAGE);
```
useMemo

GligaDumitru commented on src/pages/InvoicesPage/InvoicesPage.tsx:
```diff
@@ -12,14 +22,62 @@ const InvoiceMetadata = () => {
+    );
+  });
+  const totalPages: number = Math.ceil(filteredCustomers.length / ITEMS_PER_PAGE);
+  const paginatedItems: Array<Customer> = filteredCustomers.slice(
```
useMemo

GligaDumitru commented on src/services/api/customer.api.ts:
```diff
@@ -0,0 +1,25 @@
+
+export async function getAllCustomers() {
+  try {
+    const response = await fetch('https://68c97a0cceef5a150f650975.mockapi.io/api/v1/invoices');
```
the api should be under .env variable, not plain text

ralucamavriche committed:
refactor: update imports to use type imports from Storybook

test: add missing imports for describe, expect, and it in various test files

feat: implement debounce utility function with tests

feat: create Spinner component with stories and tests

fix: update InvoicesPage to include loading state and Spinner component

chore: remove mockCustomers data file

fix: update customer API service to use BASE_URL_API from config

chore: update TypeScript configuration for improved strictness and module resolution

chore: remove outdated Vite configuration file

ralucamavriche committed:
feat: add stories and tests for CustomerDesktopCard and CustomerMobileCard components; update InvoicesPage with loading spinner role

ralucamavriche committed:
fix: correct placeholder class syntax in SearchBar component; remove unnecessary comma in InvoicesPage story
